yo/db.py

```diff
@@ -74,16 +74,15 @@
     sa.Column('nid', sa.String(36), primary_key=True),
     sa.Column('notify_type', sa.String(20), nullable=False, index=True),
     sa.Column('to_username', sa.String(20), nullable=False, index=True),
-    sa.Column('from_username', sa.String(20), index=True, nullable=True),
-    sa.Column('json_data', sa.UnicodeText),
+    sa.Column('json_data', sa.UnicodeText(1024)),
 
     # wwwpoll specific columns
     sa.Column('created', sa.DateTime,default=sa.func.now(),nullable=False,index=True),
     sa.Column('updated', sa.DateTime, nullable=True, index=True),
     sa.Column('read', sa.Boolean(), default=False),
     sa.Column('shown', sa.Boolean(), default=False),
 
-    sa.UniqueConstraint('to_username','notify_type','json_data',name='yo_wwwpoll_idx'),
+#    sa.UniqueConstraint('to_username','notify_type','json_data',name='yo_wwwpoll_idx'),
```

Comment:
The db is responsible for the prevention of duplicate notifications, and so we must have a uniqueness constraint on the tables.

Reply:
Of course, but with the json_data field this was causing problems, since MySQL doesn't seem to support uniqueness on unicode fields.

My suggestion here would be to use the trxid from the blockchain.